elasticsearch-model/lib/elasticsearch/model/indexing.rb

```diff
@@ -393,7 +393,9 @@ def delete_document(options={})
         def update_document(options={})
           if changed_attributes = self.instance_variable_get(:@__changed_attributes)
             attributes = if respond_to?(:as_indexed_json)
-              self.as_indexed_json.select { |k,v| changed_attributes.keys.map(&:to_s).include? k.to_s }
```

Comment:
Is this change for performance? Original version seemed more readable.

Reply:
My original change included with_indifferent_access method which is a rails extension which I removed here.